JSONObject.java

```diff
@@ -1759,12 +1763,22 @@ public boolean similar(Object other) {
             return false;
         }
     }
-
+    
+    /**
+     * Tests if the value should be tried as a decimal. It makes no test if there are actual digits.
+     * 
+     * @param val value to test
+     * @return true if the string is "-0" or if it contains '.', 'e', or 'E', false otherwise.
+     */
+    protected static boolean isDecimalNotation(final String val) {
+        return val.indexOf('.') > -1 || val.indexOf('e') > -1
+                || val.indexOf('E') > -1 || "-0".equals(val);
+    }
+    
```

Comment:
Nice "quick and dirty" function, what happens if running on a server on a different locale?

Anyway the following example can help, I'm guessing:
```
  static final char MINUS_SIGN;
  static final char DECIMAL_SEPARATOR;
  
  static {
    final DecimalFormatSymbols currentLocaleSymbols = DecimalFormatSymbols.getInstance();
    MINUS_SIGN = currentLocaleSymbols.getMinusSign();
    DECIMAL_SEPARATOR = currentLocaleSymbols.getDecimalSeparator();
  }
```